src/libraries/Common/src/System/ITimer.cs

```diff
@@ -0,0 +1,27 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Threading
+{
+    /// <summary>Represents a timer that can have its due time and period changed.</summary>
+    /// <remarks>
+    /// Implementations of <see cref="Change"/>, <see cref="IDisposable.Dispose"/>, and <see cref="IAsyncDisposable.DisposeAsync"/>
+    /// must all be thread-safe such that the timer instance may be accessed concurrently from multiple threads.
+    /// </remarks>
+    public interface ITimer : IDisposable, IAsyncDisposable
+    {
+        /// <summary>Changes the start time and the interval between method invocations for a timer, using <see cref="TimeSpan"/> values to measure time intervals.</summary>
+        /// <param name="dueTime">
+        /// A <see cref="TimeSpan"/> representing the amount of time to delay before invoking the callback method specified when the <see cref="ITimer"/> was constructed.
+        /// Specify <see cref="Timeout.InfiniteTimeSpan"/> to prevent the timer from restarting. Specify <see cref="TimeSpan.Zero"/> to restart the timer immediately.
+        /// </param>
+        /// <param name="period">
+        /// The time interval between invocations of the callback method specified when the Timer was constructed.
+        /// Specify <see cref="Timeout.InfiniteTimeSpan"/> to disable periodic signaling.
+        /// </param>
+        /// <returns><see langword="true"/> if the timer was successfully updated; otherwise, <see langword="false"/>.</returns>
+        /// <exception cref="ObjectDisposedException">The timer has already been disposed.</exception>
```

Comment:
Did we want to change this, per my comment at https://github.com/dotnet/runtime/pull/83604/files#r1143400599?  Basically we'd:
- remove this and instead that just becomes a part of the return
- update Timer to always return gracefully rather than throwing ODE for Change
- look for our own catch (ObjectDisposedException) blocks around Timer.Change and remove them